gtsam/hybrid/tests/testHybridBayesNet.cpp

```diff
@@ -387,7 +387,8 @@ TEST(HybridBayesNet, Sampling) {
       std::make_shared<BetweenFactor<double>>(X(0), X(1), 0, noise_model);
   auto one_motion =
       std::make_shared<BetweenFactor<double>>(X(0), X(1), 1, noise_model);
-  std::vector<NonlinearFactor::shared_ptr> factors = {zero_motion, one_motion};
+  std::vector<std::pair<NonlinearFactor::shared_ptr, double>> factors = {
```

Comment:
Use typedef, avoid vectors

Reply:
Something to note: The DecisionTree API takes a vector, so this constructor just forwards to the DecisionTree constructor.

Also, this allows us to reduce duplication. E.g. with directly specifying DecisionTrees, we have two specify discreteKeys in 2 places:
```cpp
DiscreteKeys discreteKeys{DiscreteKey(M(0), 2)};
  HybridNonlinearFactor::Factors factors(
      discreteKeys, {{zero_motion, 0.0}, {one_motion, 0.0}});
  nfg.emplace_shared<HybridNonlinearFactor>(KeyVector{X(0), X(1)}, discreteKeys,
                                            factors);
```

There is a longstanding TODO to obtain the continuous and discrete keys from the underlying factors and decision tree directly, but until we have that, this redundancy will continue to exist.